core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -0,0 +1,372 @@
+      super(ops, table, schema, new TableScanContext());
+    }
+
+    protected PositionDeletesTableScan(
+        TableOperations ops, Table table, Schema schema, TableScanContext context) {
+      super(ops, table, schema, context);
+    }
+
+    @Override
+    protected PositionDeletesTableScan newRefinedScan(
+        TableOperations newOps, Table newTable, Schema newSchema, TableScanContext newContext) {
+      return new PositionDeletesTableScan(newOps, newTable, newSchema, newContext);
+    }
+
+    @Override
+    public CloseableIterable<ScanTaskGroup<ScanTask>> planTasks() {
+      CloseableIterable<ScanTask> scanTasks = planFiles();
+      return TableScanUtil.planTaskGroups(
+          scanTasks, targetSplitSize(), splitLookback(), splitOpenFileCost());
+    }
+
+    @Override
+    protected CloseableIterable<ScanTask> doPlanFiles() {
+      Expression rowFilter = context().rowFilter();
+      String schemaString = SchemaParser.toJson(tableSchema());
+
+      Map<Integer, PartitionSpec> transformedSpecs =
+          table().specs().entrySet().stream()
+              .map(
+                  e ->
+                      Pair.of(
+                          e.getKey(), BaseMetadataTable.transformSpec(tableSchema(), e.getValue())))
+              .collect(Collectors.toMap(Pair::first, Pair::second));
+
+      CloseableIterable<ManifestFile> deleteManifests =
+          CloseableIterable.withNoopClose(snapshot().deleteManifests(tableOps().io()));
+      CloseableIterable<CloseableIterable<ScanTask>> results =
+          CloseableIterable.transform(
+              deleteManifests,
+              m -> {
+                // Filter partitions
+                CloseableIterable<ManifestEntry<DeleteFile>> deleteFileEntries =
+                    ManifestFiles.readDeleteManifest(m, tableOps().io(), transformedSpecs)
+                        .caseSensitive(isCaseSensitive())
+                        .filterRows(rowFilter)
+                        .liveEntries();
+
+                // Filter delete file type
+                CloseableIterable<ManifestEntry<DeleteFile>> positionDeleteEntries =
+                    CloseableIterable.filter(
+                        deleteFileEntries,
+                        entry -> entry.file().content().equals(FileContent.POSITION_DELETES));
+
+                Types.StructType partitionType = Partitioning.partitionType(table());
+
+                return CloseableIterable.transform(
+                    positionDeleteEntries,
+                    entry -> {
+                      PartitionSpec spec = transformedSpecs.get(entry.file().specId());
+                      String specString = PartitionSpecParser.toJson(spec);
+                      return new PositionDeleteScanTask(
+                          entry.file().copyWithoutStats(),
+                          schemaString,
+                          specString,
+                          ResidualEvaluator.of(spec, Expressions.alwaysTrue(), isCaseSensitive()),
```

Comment:
Hm, why do we always use `Expressions.alwaysTrue()`?
Shouldn't we use `filter()`, respect `shouldIgnoreResiduals()` and have a cache of evaluators?
I wonder if there is an easy way to adapt the logic in `ManifestGroup` to also apply for delete manifests.

Reply:
Added residual evaluator, residual cache, and unit test

Reply:
Looks good.